Sage-Bionetworks/schematic
# Move API `access_token` parameter to request header

This PR pulls from some of the work done previously by @linglp in #1218 and [#1599](https://github.com/Sage-Bionetworks/sage-monorepo/pull/1599) and comments from @thomasyu888, but with modifications. It is applied to `schematic` to enhance security of the API and apps deployed on AWS that use schematic as the backend.

Synapse Access Tokens are now implemented as Bearer Authentication tokens in the request headers instead of as query parameters sent in the request body. Providing this token is required for all endpoints where it is an option, with the exception of `/manifest/generate`, where it is acceptable to not provide a token if not required for accessing resources on synapse, to preserve the existing functionality.

Tests that previously had `access_token` as a parameter were marked in #1289 for test comparison purposes. They've also been updated to pass in the token in the request header as part of this PR. All of these tests are passing bar 7, which are also failing on #1289, an analogue of `develop` with only the markers changed. I believe some API tests had been identified as having problems before @linglp's leave, these are likely among them.

Tests of the `/manifest/generate` endpoint that do not require an access token to be provided are also marked and are all passing.

Access tokens are no longer visible in the logs, confirmed on the dev instance
![image](https://github.com/Sage-Bionetworks/schematic/assets/61707471/e00b7628-0624-4c19-bee1-1e709db569c4)


### To Test

- Try the endpoints that use the access token from SwaggerUI
- Try interacting with the API now that `access_token` is no longer a query parameter and instead should be included in the header
- (note) you should be able to just pass in the auth token without the `Bearer: ` prefix when working through Swagger

Depends on #1289
